src/metabase/models/card.clj

```diff
@@ -407,13 +407,17 @@
 (defmethod serdes.base/serdes-descendants "Card" [_model-name id]
   (let [card          (db/select-one Card :id id)
         source-table  (some->  card :dataset_query :query :source-table)
-        template-tags (some->> card :dataset_query :native :template-tags vals (filter :card-id))]
+        template-tags (some->> card :dataset_query :native :template-tags vals (keep :card-id))
+        snippets      (some->> card :dataset_query :native :template-tags vals (keep :snippet-id))]
```

Comment:
What happens if someone has a query where the native query is nested one level? Or if someone joins against a parameterized native query? Seems like this wouldn't work, right? Maybe using `mbql/match` to find maps with a `:template-tags` key would be a little safer.

I don't really love having that sort of code all over the place tho. Seems like something that should go in Metabase lib. Maybe something like a `native-query-parameters` function would make sense?

Reply:
I don't understand the possible cases well enough to answer those questions. It sounds like the deps might get handled transitively in that case?

I've got several cleanups I want to do to the mishmash of MBQL wrangling in `metabase.models.serialization.util`. I've created #27003 to track that.

I still want to land this PR as-is, in order to make x.45.1 with this fix, since native questions with a snippet is a case users are likely to have, and they don't work at all in serdes at `master` or the release branch.

Reply:
Cool, let's get it in since it's still a net win, but maybe a follow-on to add tests for nested stuff would make sense too. An example of what I mean would be something like

```clj
{:database 1
 :type     :query
 :query    {:source-query {:native        "SELECT * FROM {{#2}}"
                           :template-tags {"#2" {:card-id      2
                                                 :type         :card
                                                 :display-name "Christmas Card"}}}}}
```